### PR TITLE
feature: ZENKO-1376 tests for bbClient ingestion methods

### PR DESCRIPTION
**Pull request by JianqinWang**

### CODE REVIEW

**Review by tmacro — APPROVED** [review]

File: tests/functional/lib/BackbeatClient.js [hunk 1]
```diff
@@ -73,13 +32,91 @@ describe('BackbeatClient unit tests with mock server', () => {
...
+
     it('should get raftLogs', done => {
         const destReq = backbeatClient.getRaftLog({
             LogId: '1',
         });
         return destReq.send((err, data) => {
             assert.ifError(err);
-            assert.deepStrictEqual(data, mockLogs);
+            assert.strictEqual(typeof data.info, 'object');
+            assert.strictEqual(typeof data.info.start, 'number');
```
Comment (bennettbuchanan) [inline tests/functional/lib/BackbeatClient.js]: Do we not also want to assert the values as was done with the `mockLogs`?

File: tests/utils/MockMetadataServer.js [hunk 2]
```diff
@@ -225,15 +225,15 @@ class MetadataMock {
                 bucketMd._lifeCycle);
             return res.end(dummyBucketMdObj.serialize());
         } else if
-        (/\/_\/metadata\/default\/bucket\/[a-z0-9]{3,63}/.test(req.url)) {
-            return res.end(JSON.stringify(objectList));
-        } else if
-        (/\/_\/metadata\/default\/[a-z0-9]{3,63}\/[a-z0-9]{3,63}/
+        (/\/_\/metadata\/default\/bucket\/[a-z0-9]{3,63}\/[a-z0-9]{3,63}/
```
Comment (rahulreddy) [inline tests/utils/MockMetadataServer.js]: Can you put these RegEx patterns in variables and add a one-liner comment mentioning what routes they correspond to?

**Review by rahulreddy — CHANGES_REQUESTED** [review]

File: tests/functional/lib/BackbeatClient.js [hunk 3]
```diff
@@ -73,13 +32,91 @@ describe('BackbeatClient unit tests with mock server', () => {
...
+    it('should get object metadata', done => {
+        const destReq = backbeatClient.getObjectMetadata({
+            Bucket: bucketName,
+            Key: objectName,
+        });
+        return destReq.send((err, data) => {
+            assert.ifError(err);
+            assert.strictEqual(typeof data, 'object');
+            assert(data.metadata);
+            assert.strictEqual(data.metadata, 'dogsAreGood');
```
Comment (bennettbuchanan) [inline tests/functional/lib/BackbeatClient.js:120]: 👍

File: eve/main.yml [hunk 4]
```diff
@@ -83,6 +83,12 @@ stages:
           env:
             CI: "true"
             BACKBEAT_CONFIG_FILE: "tests/config.json"
+      - ShellCommand:
+          name: run feature ingestion tests
```
Comment (philipyoo) [inline eve/main.yml:87]: 👍

**Review by philipyoo — APPROVED** [review]

File: tests/functional/lib/BackbeatClient.js [hunk 5]
```diff
@@ -83,4 +64,51 @@ describe('BackbeatClient unit tests with mock server', () => {
...
+        });
+    });
+
+    it('should get object list', done => {
+        const destReq = backbeatClient.getObjectList({
+            Bucket: bucketName,
+        });
+        return destReq.send((err, data) => {
+            assert.ifError(err);
+            data.Contents.forEach(object => {
```
Comment (rahulreddy) [inline tests/functional/lib/BackbeatClient.js]: Same thing here, use build the expected object and assert using `deepStrictEqual`

File: tests/functional/lib/BackbeatClient.js [hunk 6]
```diff
@@ -83,4 +64,51 @@ describe('BackbeatClient unit tests with mock server', () => {
...
         });
     });
+
+    it('should get bucket metadata', done => {
+        const destReq = backbeatClient.getBucketMetadata({
+            Bucket: bucketName,
+        });
+        return destReq.send((err, data) => {
+            assert.ifError(err);
+            assert.strictEqual(typeof data.acl, 'object');
```
Comment (rahulreddy) [inline tests/functional/lib/BackbeatClient.js]: I suggest building the expected object using the json and assert using the `deepStrictEqual` method. This way you are validating type and value at the same time.

File: tests/utils/MockMetadataServer.js [hunk 7]
```diff
@@ -195,9 +195,28 @@ const mockLogs = {
         ] },
     ] };
 
-const mockLogString = '\\/_\\/metadata\\/admin\\/raft_sessions\\/[\\d]*\\/' +
+// url to retrieve bucket list
+const bucketListURL = '\\/_\\/metadata\\/admin\\/raft_sessions\\/[1-8]' +
```
Comment (rahulreddy) [inline tests/utils/MockMetadataServer.js:199]: 💯

**Review by rahulreddy — CHANGES_REQUESTED** [review]
Sorry for another round of requesting changes but changing the tests to test values would be nice.

**Review by rahulreddy — APPROVED** [review]

**Comment by JianqinWang** [comment]
@bert-e approve

**Pull request merged**